pkg/controller/network/zone/controller.go

```diff
@@ -0,0 +1,192 @@
+			managed.WithRecorder(event.NewAPIRecorder(
+				mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) zone.Client
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c := conn.newClientFn(awsconfig)
+
+	return &external{kube: conn.client, client: c}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client zone.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	err := e.kube.Update(ctx, configSanitize(cr))
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdate)
+	}
+
+	if cr.Status.AtProvider.ID == "" {
+		return managed.ExternalObservation{
+			ResourceExists:    false,
+			ConnectionDetails: managed.ConnectionDetails{},
+		}, nil
+	}
+
+	res, err := e.client.GetZoneRequest(&cr.Status.AtProvider.ID).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(zone.IsErrorNoSuchHostedZone, err), errGet)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	if (cr.Spec.ForProvider.Comment != nil && res.HostedZone.Config.Comment == nil) ||
+		(cr.Spec.ForProvider.Comment == nil && res.HostedZone.Config.Comment != nil) ||
+		(*cr.Spec.ForProvider.Comment != *res.HostedZone.Config.Comment) {
+		return managed.ExternalObservation{
+			ResourceExists:    true,
+			ResourceUpToDate:  false,
+			ConnectionDetails: managed.ConnectionDetails{},
+		}, nil
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  true,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	res, err := e.client.CreateZoneRequest(cr).Send(ctx)
+
+	if res != nil {
+		cr.Status.AtProvider.Update(res.CreateHostedZoneOutput)
+	}
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha3.Zone)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	_, err := e.client.UpdateZoneRequest(&cr.Status.AtProvider.ID, cr.Spec.ForProvider.Comment).Send(ctx)
```

Comment:
We can't rely on storing things in our status, unfortunately. I think this `ID` should be stored in the external name annotation via `GetExternalName` and `SetExternalName`.